src/index.ts

```diff
@@ -1,10 +1,24 @@
+import knex from './db'; 
 import server from './server';
-import './db';
 
-server.listen({ port: 8080 }, (err, address) => {
-  if (err) {
-    console.error(err);
+async function startServer() {
+  try {
+    console.log('Testing database connection...');
+    await knex.raw('SELECT 1');
+
+    console.log('Database connected successfully!');
+    
+    server.listen({ port: 8080 }, (err, address) => {
+      if (err) {
+        console.error(err);
+        process.exit(1);
+      }
+      console.log(`Server listening at ${address}`);
+    });
+  } catch (error) {
+    console.error('Error setting up database:', error);
     process.exit(1);
   }
```

Comment:
Esse processo aqui poderia ser melhorado se trocasse o `server.listen` para `await`. Perceba que a tratativa de erros está sendo feita dentro do `try` e no `catch`, mas o `catch` aparenta só estar tratando o erro do banco de dados. Aqui vai uma sugestão pra simplificar:

```suggestion
  try {
    console.log('Testing database connection...');
    await knex.raw('SELECT 1');
    console.log('Database connected successfully!');
    
    await server.listen({ port: 8080 });
    console.log(`Server listening at http://localhost:8080`);
  } catch (error) {
    console.error(error);
    process.exit(1);
  }
```

Nesse caso perceba que eu deixei o host hardcodado na mensagem de sucesso, o que é um anti-padrão. Para resolver isso, poderia buscar `host` e `port` do env, já que esses dados costumam ficar lá.

Reply:
Assim que eu adicionar um env, já ajusto isso tbm